tests/Doctrine/Tests/ORM/Functional/OneToManyToOneOrphanRemovalTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Tests\Models\CMS\CmsPhonenumber;
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Tests a bidirectional one-to-many association mapping with orphan removal.
+ */
+class OneToManyToOneOrphanRemovalTest extends OrmFunctionalTestCase
+{
+    /** @var int */
+    protected $userId;
+
+    protected function setUp(): void
+    {
+        $this->useModelSet('rating');
+
+        parent::setUp();
+
+        $user           = new CmsUser();
+        $user->status   = 'dev';
+        $user->username = 'romanb';
+        $user->name     = 'Roman B.';
+
+        $phone1              = new CmsPhonenumber();
+        $phone1->phonenumber = '123456';
+
+        $phone2              = new CmsPhonenumber();
+        $phone2->phonenumber = '234567';
+
+        $user->addPhonenumber($phone1);
+        $user->addPhonenumber($phone2);
+
+        $this->_em->persist($user);
+        $this->_em->flush();
+
+        $this->userId = $user->getId();
+        $this->_em->clear();
+    }
+
+    public function testOrphanRemoval(): void
+    {
+        $userProxy = $this->_em->getReference(CmsUser::class, $this->userId);
+
+        $this->_em->remove($userProxy);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $query  = $this->_em->createQuery('SELECT u FROM Doctrine\Tests\Models\CMS\CmsUser u');
+        $result = $query->getResult();
+
+        self::assertCount(0, $result, 'CmsUser should be removed by EntityManager');
+
+        $query  = $this->_em->createQuery('SELECT p FROM Doctrine\Tests\Models\CMS\CmsPhonenumber p');
+        $result = $query->getResult();
+
+        self::assertCount(0, $result, 'CmsPhonenumber should be removed by orphanRemoval');
+    }
+
+    /** @group DDC-3382 */
+    public function testOrphanRemovalRemoveFromCollection(): void
+    {
+        $user = $this->_em->find(CmsUser::class, $this->userId);
+
+        $phonenumber = $user->getPhonenumbers()->remove(0);
+
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $query  = $this->_em->createQuery('SELECT p FROM Doctrine\Tests\Models\CMS\CmsPhonenumber p');
+        $result = $query->getResult();
+
+        self::assertCount(1, $result, 'CmsPhonenumber should be removed by orphanRemoval');
```

Comment:
Why? You've only updated the inverse side of the relationship here, right? According to  https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/unitofwork-associations.html#important-concepts, Doctrine should not do anything here.

Reply:
hi @greg0ire I have not done a thorough update as currently I am still trying to figure out how the whole test suite works. I have noticed....
- data sets are inside models, and then model sets are in the OrmFunctionalTestCase
- I need to work out exactly where the metadata/configuration for the app goes as per YAML in the main description. 
- Then I will try to modify the test to represent what I am trying to achieve within doctrine 
other things might be missing but I am still working my way around things